### PR TITLE
fix(core): fixes issues with virtualized list when wrapped in an element

### LINKED ISSUE #4570: Virtualized arrays of objects not rendering all items since 3.9.0 if Stack component is parent
**Describe the bug**

Since virtualized arrays were introduced in v3.9.0, we're experiencing issues rendering an array of objects though a custom component registered as a form input in sanity.conf. **This issue still happens when nesting an array of objects in a `Stack` component.**

This issue is similar to [this one](https://github.com/sanity-io/sanity/issues/4396), which was fixed for generic arrays of objects.

Here's a simplified version of our component:

```js
export const TabbedInput = (props: TabbedInputProps) => {
	[...]
	return (
		<Stack space={3} style={{position: 'relative'}} className={`tabbed-inputs-${uID}`}>
			{members.map((member) => {
				[...]
				return (
					<MemberField
						key={member.key}
						member={member}
						renderInput={renderInput}
						renderField={renderField}
						renderItem={renderItem}
						renderPreview={renderPreview} />
				)
			})}
		</Stack>
	)
}
```

By moving members out of `Stack`, the list is properly rendered.

**To Reproduce**

Nest an array of objects under a `<Stack />` component.

**Expected behavior**

All items of the array should be rendered.

**Screenshots**

This is what the component looks like, only the last two entries will show up.

image

**Which versions of Sanity are you using?**

3.12.0

**What operating system are you using?**

Macos ventura

**Which versions of Node.js / npm are you running?**

node 18, pnpm 8.6

### LINKED ISSUE #4650: Virtualized arrays of objects not rendering all items when in a group of fields
**Describe the bug**

I almost re-opened #4396, which was fixed in 3.9.1, but I think it was fixed only for the main editor ("All fields", when using groups). This bug still appears in 3.13.0 when looking at arrays of objects in a group.

FYI, other opened issue related to virtualized arrays: #4570

**To Reproduce**

1. Create a document with an array of objects
2. Create at least 4 objects in that array in one document
3. Create a group in that document and add this array to it
4. Load the document, go to the group and see that not items are properly rendered.

**Expected behavior**

All items are visible.

**Screenshots**

**Which versions of Sanity are you using?**

```
@sanity/cli (global)  3.13.0 (up to date)
@sanity/cli           3.13.0 (up to date)
@sanity/color-input    3.1.0 (up to date)
@sanity/icons          2.4.1 (up to date)
@sanity/table          1.0.1 (up to date)
@sanity/ui             1.7.0 (up to date)
@sanity/util          3.13.0 (up to date)
@sanity/uuid           3.0.2 (up to date)
@sanity/vision        3.13.0 (up to date)
sanity                3.13.0 (up to date)
```

**What operating system are you using?**

Window 10

**Which versions of Node.js / npm are you running?**

```
9.7.2
v16.20.1
```

### PR DESCRIPTION
**Pull request by binoy14**
### Description

This PR intends to fix all the issues with virtualization when wrapped in elements that have a position relative as well as arrays not displaying when inside a popover.

closes #4650 
closes #4570 

### What to review

Added a debug view for testing different scenarios and from my testing everything seems to work well from all the issues that were reported. If there should be another scenarios I should test let me know I can add it to the schema. 

https://test-studio-git-bug-sc-35080.sanity.build/test/content/input-debug;virtualizationDebug 

### Notes for release

- Fixes issues where array list would not display all items when wrapped in a custom element
- Fixes an issue where array list would not display when inside a popover

### CODE REVIEW

File: packages/sanity/src/core/form/inputs/arrays/ArrayOfObjectsInput/List/ListArrayInput.tsx [hunk 1]
```diff
@@ -111,29 +111,39 @@ export function ListArrayInput<Item extends ObjectItem>(props: ArrayOfObjectsInp
...
-      const itemOffset = parentRef.current?.offsetTop ?? 0
-      cb(scroll.scrollTop - itemOffset)
-    }
+        // This is used to calculate the offsetTop of the parent element
+        // Instead of using the `offsetTop` which will use the nearest parent with `position: relative`
+        // We pass a component that we have more control over to avoid issues when wrapped in custom component
+        const itemOffset = Math.floor(parentElementTop - containerElementTop)
 
-    handleScroll()
+        cb(scroll.scrollTop - itemOffset)
```
Comment (RitaDias) [inline packages/sanity/src/core/form/inputs/arrays/ArrayOfObjectsInput/List/ListArrayInput.tsx:131]: smol nitpick: can we rename cb to something more explicit?

File: packages/sanity/src/core/form/inputs/arrays/ArrayOfObjectsInput/List/useVirtualizerScrollInstance.tsx [hunk 2]
```diff
@@ -1,10 +1,17 @@
-import {createContext, useContext} from 'react'
+import {type MutableRefObject, createContext, useContext} from 'react'
 
 /**
  * @internal
  */
 export interface VirtualizerScrollInstance {
+  /**
+   * The parent that has the overflow scroll
+   */
```
Comment (RitaDias) [inline packages/sanity/src/core/form/inputs/arrays/ArrayOfObjectsInput/List/useVirtualizerScrollInstance.tsx:9]: We love to see it we love to see it!

**Review by RitaDias — APPROVED** [review]
One small nitpick but nothing to prevent it from ship :shipit:

File: packages/sanity/src/desk/panes/document/documentPanel/documentViews/FormView.tsx [hunk 3]
```diff
@@ -136,7 +148,7 @@ export function FormView(props: FormViewProps) {
       width={1}
     >
       <PresenceOverlay margins={margins}>
-        <Box as="form" onSubmit={preventDefault} ref={formRef}>
+        <Box as="form" onSubmit={preventDefault} ref={setRef} style={{position: 'relative'}}>
```
Comment (bjoerge) [inline packages/sanity/src/desk/panes/document/documentPanel/documentViews/FormView.tsx:151]: Not sure if this is in the a hot path performance wise, but worth moving the style to the outer scope here?

**Review by bjoerge — APPROVED** [review]
Great work!

File: packages/sanity/src/desk/panes/document/documentPanel/documentViews/FormView.tsx (hunk 3 above)
Comment (binoy14) [reply packages/sanity/src/desk/panes/document/documentPanel/documentViews/FormView.tsx:151]: Ah! that was me debugging and never removed it. That style doesn't need to be there I will remove it. great catch thanks!

**Pull request merged**